### PR TITLE
hhsuite/buildhhdb

### PR DESCRIPTION
**Pull request by vagkaratzas**

### CODE REVIEW

File: modules/ebi-metagenomics/hhsuite/buildhhdb/main.nf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    tag "$meta.id"
+    label 'process_high'
+
+    container 'microbiome-informatics/hh-suite-db-builder:1.0.0'
+
+    input:
+    tuple val(meta), path(a3m)
+
+    output:
+    tuple val(meta), path("${meta.id}"), emit: hh_db
```
Comment (chrisAta) [inline modules/ebi-metagenomics/hhsuite/buildhhdb/main.nf:11]: I'm not personally a fan of the output directory just being the meta.id, it's hard to know what it is without context... maybe add a suffix that describes what kind of data is in there, so `${meta.id}_hhsuitedb` or whatever makes most sense

File: modules/ebi-metagenomics/hhsuite/buildhhdb/meta.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1', single_end:false ]`
+  - versions:
+      type: file
+      description: File containing software versions
+      pattern: "versions.yml"
+  - hh_db:
+      type: directory
+      description: HH db directory containing all precomputed required data and index files (HMMs and context states for prefiltering)
+      pattern: "*"
```
Comment (chrisAta) [inline modules/ebi-metagenomics/hhsuite/buildhhdb/meta.yml:42]: do you need a pattern for a directory? I don't actually know haha

File: modules/ebi-metagenomics/hhsuite/buildhhdb/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,78 @@
+// TODO nf-core: Once you have added the required tests, please run the following command to build this file:
```
Comment (chrisAta) [inline modules/ebi-metagenomics/hhsuite/buildhhdb/tests/main.nf.test:1]: remove TODO

File: modules/ebi-metagenomics/hhsuite/buildhhdb/tests/main.nf.test.snap [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+                            "single_end": false
+                        },
+                        [
+                            "test_a3m_a3m.ffdata:md5,d41d8cd98f00b204e9800998ecf8427e",
+                            "test_a3m_a3m.ffindex:md5,d41d8cd98f00b204e9800998ecf8427e",
+                            "test_a3m_cs219.ffdata:md5,d41d8cd98f00b204e9800998ecf8427e",
+                            "test_a3m_cs219.ffindex:md5,d41d8cd98f00b204e9800998ecf8427e",
+                            "test_a3m_cs219.log1:md5,d41d8cd98f00b204e9800998ecf8427e",
+                            "test_a3m_hmm.ffdata:md5,d41d8cd98f00b204e9800998ecf8427e",
+                            "test_a3m_hmm.ffindex:md5,d41d8cd98f00b204e9800998ecf8427e"
```
Comment (chrisAta) [inline modules/ebi-metagenomics/hhsuite/buildhhdb/tests/main.nf.test.snap:36]: I'm pretty sure the md5sum `d41d8cd98f00b204e9800998ecf8427e` is an empty file, is this expected?

File: modules/ebi-metagenomics/hhsuite/buildhhdb/tests/main.nf.test.snap [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
```
Comment (chrisAta) [inline modules/ebi-metagenomics/hhsuite/buildhhdb/tests/main.nf.test.snap:56]: same here

**Review by chrisAta — CHANGES_REQUESTED** [review]
Nice one! Just a few minor comments

File: modules/ebi-metagenomics/hhsuite/buildhhdb/tests/main.nf.test.snap (hunk 4 above)
Comment (vagkaratzas) [reply modules/ebi-metagenomics/hhsuite/buildhhdb/tests/main.nf.test.snap:36]: Yeye, this is just the stub part of the snapshot!

File: modules/ebi-metagenomics/hhsuite/buildhhdb/meta.yml (hunk 2 above)
Comment (vagkaratzas) [reply modules/ebi-metagenomics/hhsuite/buildhhdb/meta.yml:42]: You can find all choices in nf-core modules, so I don't think there is a consensus. Better remove it

File: modules/ebi-metagenomics/hhsuite/buildhhdb/main.nf (hunk 1 above)
Comment (vagkaratzas) [reply modules/ebi-metagenomics/hhsuite/buildhhdb/main.nf:11]: Agreed, changed to ${meta.id}_hh_db

File: modules/ebi-metagenomics/hhsuite/buildhhdb/meta.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. `[ id:'sample1', single_end:false ]`
+  - a3m:
+      type: directory
+      description: Directory containing a3m format MSAs
+      pattern: "*"
```
Comment (chrisAta) [inline modules/ebi-metagenomics/hhsuite/buildhhdb/meta.yml:27]: There's another pattern for directory here I missed last time

**Review by chrisAta — APPROVED** [review]
Done looks good! Just one small final thing but I've approved it

File: modules/ebi-metagenomics/hhsuite/buildhhdb/meta.yml (hunk 6 above)
Comment (vagkaratzas) [reply modules/ebi-metagenomics/hhsuite/buildhhdb/meta.yml:27]: Which I also missed :P

**Pull request merged**